sled-agent/src/services.rs

```diff
@@ -1145,6 +1146,39 @@ impl ServiceManager {
                 let Some(info) = self.inner.sled_info.get() else {
                     return Err(Error::SledAgentNotReady);
                 };
+
+                // We want to configure the dns/install SMF service inside the
+                // zone with the list of DNS nameservers.  This will cause
+                // /etc/resolv.conf to be populated inside the zone.  To do
+                // this, we need the full list of nameservers.  Fortunately, the
+                // nameservers provide a DNS name for the full list of
+                // nameservers.
+                //
+                // Note that when we configure the dns/install service, we're
+                // supplying values for an existing property group on the SMF
+                // *service*.  We're not creating a new property group, nor are
+                // we configuring a property group on the instance.
+                let all_nameservers = info
+                    .resolver
+                    .lookup_all_ipv6(internal_dns::ServiceName::InternalDns)
+                    .await?;
+                let mut dns_config_builder =
+                    PropertyGroupBuilder::new("install_props");
+                for ns_addr in &all_nameservers {
+                    dns_config_builder = dns_config_builder.add_property(
+                        "nameserver",
+                        "net_address",
+                        &ns_addr.to_string(),
+                    );
+                }
+                let dns_install = ServiceBuilder::new("network/dns/install")
+                    .add_property_group(dns_config_builder)
+                    // We do need to enable the default instance of the
+                    // dns/install service.  It's enough to just mention it
+                    // here, as the ServiceInstanceBuilder always enables the
+                    // instance being added.
+                    .add_instance(ServiceInstanceBuilder::new("default"));
+
```

Comment:
This has been fully copy-pasted from the CockroachDB initialization, right? Can we split this into a helper function to avoid duplication?